### DISCUSSION TITLE
How to remove text items in completion menu

### DISCUSSION THREAD

**Discussion by emnnipal** [LazyVim/LazyVim#2458] [Q&A] [answered]
Hi everyone, I would like to ask for help on how to remove the text items from the completion menu.

I have already checked these links regarding the same matter, but I can't figure out how to remove the spell in LazyVim:

https://www.reddit.com/r/neovim/comments/wf5dsw/nvimcmplsp_how_to_remove_text_results_from/
https://www.reddit.com/r/neovim/comments/v2ifpb/text_fields_on_lsp_auto_completion/

Thanks in advance!

**Comment by dpetka2001** [comment]
I don't understand what you want to do. Can you provide more information and a screenshot about what you'd like to achieve? LazyVim doesn't have anything about `spell` in auto-completion menu.

**Reply by emnnipal** [reply]
These types of items appear in the completion menu.

<img width="229" alt="image" src="https://github.com/LazyVim/LazyVim/assets/22177776/11bedead-be64-4536-bcef-a92f11d55419">

While the screenshot may only show a few, there are cases where the completion menu contains a lot of text like the one in the reddit post.

**Reply by dpetka2001** [reply, marked as answer]
Those come either from the `cmp-buffer` source or from the LSP source. Do you want to disable them completely? Then you wouldn't get any completions regarding text in the buffer. You could either disable auto-completion when you're inside a comment section or disable the `Text` kind all together. Try the following, but keep in mind you won't be getting any auto-completions about `Text` any more
```lua
  {
    "hrsh7th/nvim-cmp",
    opts = function(_, opts)
      -- local context = require("cmp.config.context")
      -- opts.enabled = function()
      --   return not context.in_treesitter_capture("comment")
      -- end
      opts.sources = vim.tbl_filter(function(source)
        return not vim.tbl_contains({ "buffer", "nvim_lsp" }, source.name)
      end, opts.sources)
      table.insert(opts.sources, 1, {
        name = "nvim_lsp",
        entry_filter = function(entry, _)
          return require("cmp.types").lsp.CompletionItemKind[entry:get_kind()] ~= "Text"
        end,
      })
    end,
  },
```
The commented code snippet is for disabling the auto-completion menu from popping up when inside a comment section. Not sure if you also want that.

**Reply by emnnipal** [reply]
Thank you so much, @dpetka2001. I appreciate the help.

I'll take some time to observe whether I prefer hiding the `Text` exclusively within the comment section or across all scenarios. At times, these `Text` items can be a bit bothersome, but I'll observe first.
